crates/subspace-core-primitives/src/lib.rs

```diff
@@ -131,6 +132,55 @@ impl AsRef<[u8]> for Signature {
     }
 }
 
+/// A Ristretto Schnorr signature as bytes produced by `schnorrkel` crate.
+#[derive(Copy, Clone, PartialEq, Eq, Ord, PartialOrd, Hash, Encode, Decode, TypeInfo)]
+#[cfg_attr(feature = "std", derive(Debug))]
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+pub struct LocalChallenge(
+    #[cfg_attr(feature = "std", serde(with = "serde_arrays"))] [u8; SIGNATURE_LENGTH],
+);
+
+impl Default for LocalChallenge {
+    fn default() -> Self {
+        Self([0u8; SIGNATURE_LENGTH])
+    }
+}
+
+impl From<[u8; SIGNATURE_LENGTH]> for LocalChallenge {
+    fn from(bytes: [u8; SIGNATURE_LENGTH]) -> Self {
+        Self(bytes)
+    }
+}
+
+impl From<LocalChallenge> for [u8; SIGNATURE_LENGTH] {
+    fn from(signature: LocalChallenge) -> Self {
+        signature.0
+    }
+}
+
+impl Deref for LocalChallenge {
+    type Target = [u8];
+
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+impl AsRef<[u8]> for LocalChallenge {
+    fn as_ref(&self) -> &[u8] {
+        &self.0
+    }
+}
+
+impl LocalChallenge {
+    /// Derive tags search target from local challenge.
+    pub fn derive_target(&self) -> Tag {
+        crypto::sha256_hash(&self.0)[..TAG_SIZE]
+            .try_into()
+            .expect("Signature is always bigger than tag; qed")
```

Comment:
Why not return the final target `u64`? The doc comment is kind of inconsistent as it implies it should return a target which I understand is a number `u64` but return a `Tag`.

Reply:
We only interpret it as u64 during tags database lookup. Tag should eventually also be a new type wrapper.

Reply:
Then it might be better to have two methods `derive_target` and `derive_tag` where `derive_target` simply calls `derive_tag` to make the tag and target concept more discriminative.

Reply:
The search space is essentially a 64-bit hash ring, so this is why both tags and search target have type `Tag`. I'm not sure I'm following how two methods would help.

Reply:
Okay, so the target and any tag are essentially a point on the ring, right? If so, I'll just say `Tag` is probably not a good name in that regard :).

Reply:
Yeah, do you have better option in mind? Not like this PR invented it though.

Reply:
Not now, but I believe we'll come up with some better names eventually :D.